packages/ember-metal/lib/property_set.js

```diff
@@ -67,7 +67,7 @@ export function set(obj, keyName, value, tolerant) {
   }
 
   Ember.assert("You need to provide an object and key to `set`.", !!obj && keyName !== undefined);
-  Ember.assert('calling set on destroyed object', !obj.isDestroyed);
+  Ember.assert("calling set('" + keyName + "') on a destroyed object", !obj.isDestroyed);
```

Comment:
can you also add 'obj' to the output. That will force its toString to be put into the assertion, which might help even more.